torch/csrc/api/include/torch/nn/functional/loss.h

```diff
@@ -26,6 +26,35 @@ inline Tensor cosine_embedding_loss(
       input1, input2, target, options.margin(), options.reduction());
 }
 
+inline Tensor _smooth_l1_loss(const Tensor& input, const Tensor& target) {
+    auto t = torch::abs(input - target);
+    return torch::where(t < 1, 0.5 * torch::pow(t, 2), t - 0.5);
+}
+
+inline Tensor smooth_l1_loss(
+    const Tensor& input,
+    const Tensor& target,
+    const SmoothL1LossOptions& options = {}) {
+  if (target.sizes() != input.sizes()) {
+    TORCH_WARN("Using a target size (", target.sizes(), ") that is different to the input size (", input.sizes(), "). ",
+                  "This will likely lead to incorrect results due to broadcasting. ",
+                  "Please ensure they have the same size.");
```

Comment:
nit: would be awesome to align these two lines with the first line in `TORCH_WARN` :D